### DISCUSSION TITLE
am i retarded?

### DISCUSSION THREAD

**Discussion by FriGiN** [Mydayyy/Valheim-ServerSideMap#26] [Q&A] [answered]
I can't get this to work right, the .explored is being created in the server directory, however pins, and map sharing is not working.

I have put the .dll's you requested everywhere I can think of:

\Valheim dedicated server\
\Valheim dedicated server\BepInEx\
\Valheim dedicated server\BepInEx\plugins\

from the source code it say they should be in ..\Libs\ so i assume that is the \BepInEx\ folder. Am I just really stupid and cant figure this out?

**Comment by Mydayyy** [comment]
Are you running it on the client too?

Cheers
Mydayyy

**Reply by Mydayyy** [reply]
Also, did you install bepinex?

The Libs folder is only relevant for development for this plugin

**Reply by FriGiN** [reply]
I am indeed, 2 of us are doing it... i think i had a eureka right there as i was typing i need to copy the listed dll's on client and server side to ..\Libs\ which should be the \BepInEx\Libs directory right?

**Reply by Mydayyy** [reply]
not quite, Libs is only relevant for development. Plugins is the correct location. Given that the .explored file is created on the server we can assume that the server is working and loading the plugin correctly.

That would leave the client not loading the plugin correctly. Can you confirm that it is loaded via the bepinex logfiles?

Correct path: Valheim/BepInEx/plugins/ServerSideMap.dll

**Reply by FriGiN** [reply]
[Info   :   BepInEx] Loading [ServerSideMap 1.2.1.0]
[Info   :  HarmonyX] ### Harmony id=eu.mydayyy.plugins.serversidemap, version=2.3.2.0, location=E:\SteamLibrary\steamapps\common\Valheim\BepInEx\core\0Harmony.dll, env/clr=4.0.30319.42000, platform=Win32NT
### Started from static void HarmonyLib.Tools.Logger::Log(HarmonyLib.Tools.Logger+LogChannel channel, System.Func<string> message), location E:\SteamLibrary\steamapps\common\Valheim\BepInEx\core\0Harmony.dll
### At 2021-03-09 10.01.32
[Info   :  HarmonyX] ### Harmony id=harmony-auto-b7cbf9f1-dbc7-4223-a9ce-f6af7b429fea, version=2.3.2.0, location=E:\SteamLibrary\steamapps\common\Valheim\BepInEx\core\0Harmony.dll, env/clr=4.0.30319.42000, platform=Win32NT
### Started from static void HarmonyLib.Tools.Logger::Log(HarmonyLib.Tools.Logger+LogChannel channel, System.Func<string> message), location E:\SteamLibrary\steamapps\common\Valheim\BepInEx\core\0Harmony.dll
### At 2021-03-09 10.01.32
[Info   :  HarmonyX] Patching void Minimap::Explore(UnityEngine.Vector3 p, float radius) with 0 prefixes, 1 postfixes, 0 transpilers, 0 finalizers
1 postfixes:
* static void ServerSideMap.ExplorationDatabase+MinimapPatchExploreInterval::Postfix(Minimap __instance)

[Info   :  HarmonyX] Running ILHook manipulator on void Minimap::Explore(UnityEngine.Vector3 p, float radius)
[Info   :  HarmonyX] Writing postfixes
[Info   :  HarmonyX] Patching bool Minimap::Explore(int x, int y) with 0 prefixes, 1 postfixes, 0 transpilers, 0 finalizers
1 postfixes:
* static void ServerSideMap.ExplorationMapSync+MinimapPatchExplore::Postfix(int x, int y, bool __result)

[Info   :  HarmonyX] Running ILHook manipulator on bool Minimap::Explore(int x, int y)
[Info   :  HarmonyX] Writing postfixes
[Info   :  HarmonyX] Patching void Minimap::SetMapData(byte[] data) with 1 prefixes, 1 postfixes, 0 transpilers, 0 finalizers
1 prefixes:
* static void ServerSideMap.ExplorationMapSync+MinimapPatchSetMapData::Prefix()
1 postfixes:
* static void ServerSideMap.ExplorationMapSync+MinimapPatchSetMapData::Postfix(Minimap __instance)

[Info   :  HarmonyX] Running ILHook manipulator on void Minimap::SetMapData(byte[] data)
[Info   :  HarmonyX] Writing prefixes
[Info   :  HarmonyX] Writing postfixes
[Info   :  HarmonyX] Patching void ZNet::RPC_PeerInfo(ZRpc rpc, ZPackage pkg) with 0 prefixes, 1 postfixes, 0 transpilers, 0 finalizers
1 postfixes:
* static void ServerSideMap.InitialMapSync+ZnetPatchRPC_PeerInfo::Postfix(ZRpc rpc, ZNet __instance)

[Info   :  HarmonyX] Running ILHook manipulator on void ZNet::RPC_PeerInfo(ZRpc rpc, ZPackage pkg)
[Info   :  HarmonyX] Writing postfixes
[Info   :  HarmonyX] Patching void Minimap::SetMapData(byte[] data) with 1 prefixes, 2 postfixes, 0 transpilers, 0 finalizers
1 prefixes:
* static void ServerSideMap.ExplorationMapSync+MinimapPatchSetMapData::Prefix()
2 postfixes:
* static void ServerSideMap.ExplorationMapSync+MinimapPatchSetMapData::Postfix(Minimap __instance)
* static void ServerSideMap.InitialMapSync+MinimapPatchSetMapData::Postfix(Minimap __instance)

[Info   :  HarmonyX] Running ILHook manipulator on void Minimap::SetMapData(byte[] data)
[Info   :  HarmonyX] Writing prefixes
[Info   :  HarmonyX] Writing postfixes
[Info   :  HarmonyX] Patching void Minimap::ClearPins() with 0 prefixes, 1 postfixes, 0 transpilers, 0 finalizers
1 postfixes:
* static void ServerSideMap.InitialPinSync+MinimapPatchClearPins::Postfix(Minimap __instance)

[Info   :  HarmonyX] Running ILHook manipulator on void Minimap::ClearPins()
[Info   :  HarmonyX] Writing postfixes
[Info   :  HarmonyX] Patching void Minimap::SetMapData(byte[] data) with 1 prefixes, 3 postfixes, 0 transpilers, 0 finalizers
1 prefixes:
* static void ServerSideMap.ExplorationMapSync+MinimapPatchSetMapData::Prefix()
3 postfixes:
* static void ServerSideMap.ExplorationMapSync+MinimapPatchSetMapData::Postfix(Minimap __instance)
* static void ServerSideMap.InitialMapSync+MinimapPatchSetMapData::Postfix(Minimap __instance)
* static void ServerSideMap.InitialPinSync+MinimapPatchSetMapData::Postfix(Minimap __instance)

[Info   :  HarmonyX] Running ILHook manipulator on void Minimap::SetMapData(byte[] data)
[Info   :  HarmonyX] Writing prefixes
[Info   :  HarmonyX] Writing postfixes
[Info   :  HarmonyX] Patching void ZNet::RPC_PeerInfo(ZRpc rpc, ZPackage pkg) with 0 prefixes, 2 postfixes, 0 transpilers, 0 finalizers
2 postfixes:
* static void ServerSideMap.InitialMapSync+ZnetPatchRPC_PeerInfo::Postfix(ZRpc rpc, ZNet __instance)
* static void ServerSideMap.InitialPinSync+ZnetPatchRPC_PeerInfo::Postfix(ZRpc rpc, ZNet __instance)

[Info   :  HarmonyX] Running ILHook manipulator on void ZNet::RPC_PeerInfo(ZRpc rpc, ZPackage pkg)
[Info   :  HarmonyX] Writing postfixes
[Info   :  HarmonyX] Patching void Minimap::ShowPinNameInput(Minimap+PinData pin) with 0 prefixes, 1 postfixes, 0 transpilers, 0 finalizers
1 postfixes:
* static void ServerSideMap.PinSync+MinimapPatchShowPinNameInput::Postfix(Minimap __instance, Minimap+PinData pin)

[Info   :  HarmonyX] Running ILHook manipulator on void Minimap::ShowPinNameInput(Minimap+PinData pin)
[Info   :  HarmonyX] Writing postfixes
[Info   :  HarmonyX] Patching void Minimap::OnMapLeftClick() with 0 prefixes, 1 postfixes, 0 transpilers, 0 finalizers
1 postfixes:
* static void ServerSideMap.PinSync+MinimapPatchOnMapLeftClick::Postfix(Minimap __instance)

[Info   :  HarmonyX] Running ILHook manipulator on void Minimap::OnMapLeftClick()
[Info   :  HarmonyX] Writing postfixes
[Info   :  HarmonyX] Patching Minimap+PinData Minimap::GetClosestPin(UnityEngine.Vector3 pos, float radius) with 0 prefixes, 1 postfixes, 0 transpilers, 0 finalizers
1 postfixes:
* static void ServerSideMap.PinSync+MinimapPatchGetClosestPin::Postfix(Minimap __instance, Minimap+PinData& __result, UnityEngine.Vector3 pos, float radius)

[Info   :  HarmonyX] Running ILHook manipulator on Minimap+PinData Minimap::GetClosestPin(UnityEngine.Vector3 pos, float radius)
[Info   :  HarmonyX] Writing postfixes
[Info   :  HarmonyX] Patching void Minimap::RemovePin(Minimap+PinData pin) with 0 prefixes, 1 postfixes, 0 transpilers, 0 finalizers
1 postfixes:
* static void ServerSideMap.PinSync+MinimapPatchRemovePin::Postfix(Minimap __instance, Minimap+PinData pin)

[Info   :  HarmonyX] Running ILHook manipulator on void Minimap::RemovePin(Minimap+PinData pin)
[Info   :  HarmonyX] Writing postfixes
[Info   :  HarmonyX] Patching void Minimap::UpdateNameInput() with 0 prefixes, 1 postfixes, 0 transpilers, 0 finalizers
1 postfixes:
* static void ServerSideMap.PinSync+MinimapPatchUpdateNameInput::Postfix(Minimap __instance, Minimap+PinData ___m_namePin)

[Info   :  HarmonyX] Running ILHook manipulator on void Minimap::UpdateNameInput()
[Info   :  HarmonyX] Writing postfixes
[Info   :  HarmonyX] Patching void ZNet::LoadWorld() with 0 prefixes, 1 postfixes, 0 transpilers, 0 finalizers
1 postfixes:
* static void ServerSideMap.SaveWorld+ZnetPatchLoadMap::Postfix(ZNet __instance)

[Info   :  HarmonyX] Running ILHook manipulator on void ZNet::LoadWorld()
[Info   :  HarmonyX] Writing postfixes
[Info   :  HarmonyX] Patching void ZNet::SaveWorldThread() with 0 prefixes, 1 postfixes, 0 transpilers, 0 finalizers
1 postfixes:
* static void ServerSideMap.SaveWorld+ZnetPatchSaveWorldThread::Postfix()

[Info   :  HarmonyX] Running ILHook manipulator on void ZNet::SaveWorldThread()
[Info   :  HarmonyX] Writing postfixes
[Info   :  HarmonyX] Patching void ZNet::OnNewConnection(ZNetPeer peer) with 0 prefixes, 1 postfixes, 0 transpilers, 0 finalizers
1 postfixes:
* static void ServerSideMap.ServerSideMap+ZnetPatchOnNewConnection::Postfix(ZNetPeer peer, ZNet __instance)

[Info   :  HarmonyX] Running ILHook manipulator on void ZNet::OnNewConnection(ZNetPeer peer)
[Info   :  HarmonyX] Writing postfixes
[Info   :  HarmonyX] Patching void Minimap::Awake() with 0 prefixes, 1 postfixes, 0 transpilers, 0 finalizers
1 postfixes:
* static void ServerSideMap._Minimap+Awake::Postfix(Minimap __instance)

[Info   :  HarmonyX] Running ILHook manipulator on void Minimap::Awake()
[Info   :  HarmonyX] Writing postfixes
[Info   :  HarmonyX] Patching void ZNet::Awake() with 0 prefixes, 1 postfixes, 0 transpilers, 0 finalizers
1 postfixes:
* static void ServerSideMap._ZNet+Awake::Postfix(ZNet __instance)

**Comment by JTF195** [comment]
Assuming BepInEx itself is installed correctly on the client and server, all you should need are `<path>/Valheim/BepInEx/Plugins/ServerSideMap.dll` on the client, and `<path>/Valheim dedicated server/BepInEx/Plugins/ServerSideMap.dll` on the server

**Comment by FriGiN** [comment, marked as answer]
I was not retarded, i was complicating things thinking the problem was me, turns out the monkey that was helping me test, well he put the .dll into \Valheim\ directory even though i told him what to do.   Thanks guys!

**Reply by Mydayyy** [reply]
Glad to hear its resolved :D have fun

Cheers
Mydayyy
